test/e2e/node_feature_discovery_test.go

```diff
@@ -854,6 +869,45 @@ func checkForNodeLabels(cli clientset.Interface, expectedNewLabels map[string]k8
 	return err
 }
 
+// checkForNfdNodeAnnotations waits and checks that node is annotated as expected.
+func checkForNodeAnnotations(cli clientset.Interface, expectedNewAnnotations map[string]k8sAnnotations, oldNodes []corev1.Node) error {
```

Comment:
func name does not match comment

Reply:
Also what is the diff from this func vs `waitForNfdNodeAnnotations`

Reply:
`waitForNfdNodeAnnotations` only check whether the expectd k,v  on node annotations.But the node annotaion not contains in expected will miss compare.
`checkForNodeAnnotations` will compare the whole node annotation obj.